src/device_impl.rs

```diff
@@ -64,98 +64,62 @@ where
 {
     /// Initialize registers
     pub fn init(&mut self) -> Result<(), Error<CommE, PinE>> {
-        let temp_cfg_reg = self
-            .temp_cfg_reg_a
-            .with_high(BF::TEMP_EN0)
-            .with_high(BF::TEMP_EN1);
-        self.iface
-            .write_accel_register(Register::TEMP_CFG_REG_A, temp_cfg_reg.bits)?;
-        self.temp_cfg_reg_a = temp_cfg_reg;
         let reg4 = self.ctrl_reg4_a.with_high(BF::ACCEL_BDU);
         self.iface
             .write_accel_register(Register::CTRL_REG4_A, reg4.bits)?;
         self.ctrl_reg4_a = reg4;
+
         let regc = self.cfg_reg_c_m.with_high(BF::MAG_BDU);
         self.iface
             .write_mag_register(Register::CFG_REG_C_M, regc.bits)?;
         self.cfg_reg_c_m = regc;
+
+        Ok(())
+    }
+
+    /// Enable the temperature sensor.
+    #[inline]
+    fn enable_temperature_sensor(&mut self) -> Result<(), Error<CommE, PinE>> {
+        if self.temp_cfg_reg_a.is_high(BF::TEMP_EN) {
```

Comment:
I would rather not ask for the status here and always overwrite the registers. This would solve the problem of a un-synchronized status of the `temp_cfg_reg_a` variable in the driver, for example due to a hardware reset of the device. The price of a maybe unnecessary bus transaction would just be following user commands.

Reply:
Reverted this change again.